### ISSUE TITLE
kvm machine type images don't boot out of the box in s390x systems

### ISSUE THREAD

**Issue by gmoro** [OSInside/kiwi#2247]
## Problem description
If not explicit instructed to load the qemu dracut module grub2-zipl-setup will create a --host-only initrd, that will generate an initrd with the necessary virtio kernel modules in it.

Although this could be done with custom config.sh scripts, for the specific machine type we should probably do this automatically in kiwi

Also, grub2-zipl-setup is exclusive to SUSE, so it should be ignored for other systems.
## Expected behaviour
Regular kiwi file with kvm type built on s390x generates a bootable image
## Steps to reproduce the behaviour
Create a kiwi defition with 
`<type image="oem" bootkernel="custom" bootpartition="true" bootpartsize="300" bootfilesystem="ext2" bootloader="grub2_s390x_emu" btrfs_quota_groups="true" btrfs_root_is_snapshot="true"filesystem="btrfs" format="qcow2" kernelcmdline="hvc_iucv=8 TERM=dumb">
`

Including the packages:
`
       <package name="grub2-s390x-emu"/>
       <package name="blog"/>
`

**Comment by schaefi** [comment]
I think a good way to solve this is to extend ```grub2-zipl-setup``` by an option such that it includes the needed dracut modules. imho the only acceptable way from a kiwi point of view. This will allow us to modify the respective code from here: 

* https://github.com/OSInside/kiwi/blob/master/kiwi/bootloader/install/grub2.py#L197

Maybe you can talk to the maintainers of ```grub2-zipl-setup``` to support the image use case better ?

Thanks much

**Comment by gmoro** [comment]
Agreed, we talked about this and other issues and we might have to extend the s390x-tools but also kiwi can possibly let the system know it's a KVM image? This would not impact in RAW images that don't need the virtio kernel modules at all

**Comment by schaefi** [comment]
> but also kiwi can let the system know it's a KVM image

you think something like an image ID ? This already exists as

```xml
<image ... id="some-free-form-identification" ...>
    ...
</image>
```

This will cause kiwi to create a file ```/etc/ImageID``` which you could consume as part of the s390x-tools

Would that be sufficient ?

**Comment by sanmuny** [comment]
hi @gmoro , I met the same problem. Do you know how to make a workaround with custom config.sh scripts?

**Comment by schaefi** [comment]
closing due to no further information
